src/components/FileBrowser.vue

```diff
@@ -0,0 +1,285 @@
+<script>
+import { sizeFormatter } from '../utils/mixins';
+
+const GIRDER_FOLDER_ENDPOINT = 'folder';
+const GIRDER_ITEM_ENDOINT = 'item';
+const ICON_MAP = {
+  folder: 'folder',
+  item: 'file',
+};
+
+export default {
+  props: {
+    // location should have properties `type`, `id`
+    location: {
+      type: Object,
+      required: true,
+    },
+    selectEnabled: {
+      type: Boolean,
+      default: true,
+    },
+    multiSelectEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    uploadEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    newItemEnabled: {
+      type: Boolean,
+      default: true,
+    },
+    newFolderEnabled: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  mixins: [sizeFormatter],
+  inject: ['girderRest'],
+  data() {
+    return {
+      rowsLoading: false,
+      breadcrumbLoading: false,
+      pagination: {
+        rowsPerPage: 10,
+        page: 1,
+      },
+      selected: [],
+      refreshCounter_: 0, // https://github.com/girder/girder_web_components/issues/39
+    };
+  },
+  asyncComputed: {
```

Comment:
This is a minor thing, I don't know if it is just me, maybe we should place asyncComputed after computed in general?